Gillian-C/lib/Chunk.ml

```diff
@@ -7,9 +7,30 @@ type t = Compcert.AST.memory_chunk =
   | Mint64
   | Mfloat32
   | Mfloat64
-  | Many32
-  | Many64
-[@@deriving yojson]
+  | Mptr
+[@@deriving eq, yojson]
+
+let of_compcert : Compcert.AST.memory_chunk -> t = function
+  | Mint8signed -> Mint8signed
+  | Mint8unsigned -> Mint8unsigned
+  | Mint16signed -> Mint16signed
+  | Mint16unsigned -> Mint16unsigned
+  | Mint32 -> Mint32
+  | Mint64 -> Mint64
+  | Mfloat32 -> Mfloat32
+  | Mfloat64 -> Mfloat64
+  | Many32 | Many64 -> Mptr (* Should we fail with error here instead? *)
```

Comment:
Sorry more comments
Yes, that should fail here, but also, what is `of_compcert` actually doing?
To convert from compcert to your new kind, you need the context?

Reply:
`of_compcert` is needed when translating the compcert chunks from the CSM program into out internal Chunk representation, called during `ELoad` and `SStore` functions.